pkg/metrics/gatherer/gatherer.go

```diff
@@ -0,0 +1,110 @@
+package gatherer
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/openshift/installer/pkg/metrics/builder"
+	"github.com/openshift/installer/pkg/metrics/pushclient"
+)
+
+var (
+	// metricRegistry holds all the MetricBuilder objects for all the metrics that have been designed
+	// and ready for extraction from the installer.
+	metricRegistry = make(map[string]*builder.MetricBuilder)
```

Comment:
This entire package is designed to be global.

I want us to implement this as an object that is exposed globally for ease today because we don't have a great to provide it as dependency in our assets.

similar to http.Client... there is an explicit http.Client in stdlib but also a global empty client for calling http.Get directly.